api/spec/src/productcatalog/subscription.tsp

```diff
@@ -177,11 +177,58 @@ model SubscriptionItem {
 }
 
 /**
- * Create subscription request body.
+ * Create a subscription.
  */
 @friendlyName("CreateSubscriptionRequestBody")
-model CreateSubscriptionRequestBody {
+@oneOf
+union CreateSubscriptionRequestBody {
```

Comment:
This'd be the name generated by`Rest.Resource.ResourceCreateModel<Subscription>`. I know it doesn't support unions, but we should follow the same naming if possible.

```suggestion
union SubscriptionCreate {
```